exercise-app.js

```diff
@@ -28,13 +19,59 @@ const Weather = ({ temp, description, icon }) => (
 );
 
 
-const App = () => (
+class App extends React.Component {
+
+  constructor(props) {                // where do we hook up these props to ? where does this prop data come from?
+  super(props);                       // what is the purpose of this? why do we need to call super?
+  this.state = {
+  temp: 0,
+  description: 'not loaded yet',
+  icon: `http://openweathermap.org/img/w/10d.png`};  // always intialize a state? is the state to be updated by passing in props?
+  // this.reshapeData = this.reshapeData.bind(this)  // when do we need to bind and when not (I've seen both...)
+  this.render()                       // was i supposed call render here? (I didn't and the card didn't show for a bit upon load)
+}
+
+ getWeather() {
+    return fetch(`http://api.openweathermap.org/data/2.5/weather?id=6167865&appid=581522ec0b69c0bcc097344d84601245&units=metric`)
+      .then(res => {
+        // console.log(37,res && res.json())     // when I uncomment this... I get error "promise already used" why?
```

Comment:
`res.json()` returns a promise. If you call it once while logging and then call it again for the return value then you are trying to use it twice. Hence then `"promise already used"` error. If you want to log it then try this instead:

```js
return fetch(`http://api.openweathermap.org/data/2.5/weather?id=6167865&appid=581522ec0b69c0bcc097344d84601245&units=metric`)
  .then(res => res.json())
  .then(data => {
    console.log(data);
    return data
  }
  .then( ... ) 
```